server.go

```diff
@@ -5322,6 +5338,33 @@ func (s *server) SendCustomMessage(peerPub [33]byte, msgType lnwire.MessageType,
 	return peer.SendMessageLazy(true, msg)
 }
 
+// SendOnionMessage sends a custom message to the peer with the specified
+// pubkey.
+// TODO(gijs): change this message to include path finding.
+func (s *server) SendOnionMessage(peerPub [33]byte,
+	blindingPoint *btcec.PublicKey, onion []byte) error {
+
+	peer, err := s.FindPeerByPubStr(hex.EncodeToString(peerPub[:]))
+	if err != nil {
+		return err
+	}
+
+	// We'll wait until the peer is active.
+	select {
+	case <-peer.ActiveSignal():
+	case <-peer.QuitSignal():
```

Comment:
Isn't the quitsignal set if the peer is not active ?

Reply:
Again, see `SendCustomMessage` above. I just followed that pattern. Is there a reason to do it differently here? Or is it time to refactor both?